### PR TITLE
feat!: remove `@vitest/coverage-c8` package

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Continues #3339
- Removes `@vitest/coverage-c8` package
- Removes all mentions of `c8` from documentation
- Throw informative error when `coverage.provider: 'c8'` is used

### BREAKING CHANGES:

- `coverage.provider: 'c8'` is no longer supported. Use `coverage.provider: 'v8'` instead. See https://github.com/vitest-dev/vitest/pull/3339.

### Actions after release

- Deprecate `@vitest/coverage-c8` NPM package: https://docs.npmjs.com/cli/v8/commands/npm-deprecate

### CODE REVIEW

File: docs/guide/coverage.md
Comment (AriPerkkio) [inline docs/guide/coverage.md:1]: This guide still mentions `c8` through coverage ignore hints. That is correct syntax. 

https://github.com/vitest-dev/vitest/blob/1e8760ab2b4ae02288ebaf0ab27f9ed314d41980/docs/guide/coverage.md?plain=1#L160-L164

I've opened a feature request to changes this: https://github.com/istanbuljs/v8-to-istanbul/issues/215

**Comment by AriPerkkio** [comment]
After this one is merged and released, should we add deprecation warning to `@vitest/coverage-c8@0.32.x` NPM package? https://docs.npmjs.com/cli/v8/commands/npm-deprecate

**Comment by sheremet-va** [comment]
> After this one is merged and released, should we add deprecation warning to `@vitest/coverage-c8@0.32.x` NPM package? [docs.npmjs.com/cli/v8/commands/npm-deprecate](https://docs.npmjs.com/cli/v8/commands/npm-deprecate)

Yes. You can add a section in the description for what to do here after we released the new major. We should also probably add a filter for c8 package to stop releasing new versions (we run with `pnpm release` which is a script in `package.json`)

**Comment by sheremet-va** [comment]
Can you resolve conflicts with the lockfile?

**Comment by AriPerkkio** [comment]
Rebased `main` and resolved lockfiles. Good to go. 🚀

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**